webdocs/docs/setup.md

```diff
@@ -0,0 +1,457 @@
+```toml
+internal_secret = "sample_secret_12345678901234567890"
+```
+
+Next, we need to tell the core about our authenticaton method:
+```toml
+[[global.auth_methods]]
+tag = "auth_test"
+name = "Test Authenticator"
+image_path = ""
+start = "http://localhost:8001"
+```
+
+and our communication method:
+```
+[[global.comm_methods]]
+tag = "comm_test"
+name = "Test Communicator"
+image_path = ""
+start = "http://localhost:8002"
+
+```
+If we wanted to add more of these, we would just repeat with more plugin blocks. We leave the image paths empty here as we won't need them for our testing.
+
+Finally, we need to specify our purpose and how its flow should look:
+```toml
+[[global.purposes]]
+tag = "test_purpose"
+attributes = [ "test_attribute" ]
+allowed_auth = [ "auth_test" ]
+allowed_comm = [ "comm_test" ]
+```
+
+Combined, this gives as content of `config.toml`:
+```toml
+[global]
+server_url = "http://localhost:8000"
+internal_url = "http://localhost:8000"
+port = 8000
+internal_secret = "sample_secret_12345678901234567890"
+
+[[global.auth_methods]]
+tag = "auth_test"
+name = "Test Authenticator"
+image_path = ""
+start = "http://localhost:8001"
+
+[[global.comm_methods]]
+tag = "comm_test"
+name = "Test Communicator"
+image_path = ""
+start = "http://localhost:8002"
+
+[[global.purposes]]
+tag = "test_purpose"
+attributes = [ "test_attribute" ]
+allowed_auth = [ "auth_test" ]
+allowed_comm = [ "comm_test" ]
+```
+
+We are now ready to start the core with
+```bash
+ROCKET_CONFIG=config.toml cargo run
+```
+which will build and then run the core with our above configuration.
+
+## Testing
+
+We can now test our setup by opening a 4th terminal window and using [httpie](https://httpie.io/) to make a test request to the core with the intent to start a session with the testing plugins for our test_purpose:
+```bash
+http POST http://localhost:8000/start purpose=test_purpose auth_method=auth_test comm_method=comm_test
+```
+
+This will show our server response, containing the client url:
+```http
+HTTP/1.1 200 OK
+content-length: 149
+content-type: application/json
+date: Tue, 13 Apr 2021 12:48:40 GMT
+server: Rocket
+
+{
+    "client_url": "http://localhost:8001/browser/WyJ0ZXN0X2F0dHJpYnV0ZSJd/aHR0cDovL2xvY2FsaG9zdDo4MDAyL3Vp/aHR0cDovL2xvY2FsaG9zdDo4MDAyL2F1dGhfcmVzdWx0"
+}
+
+```
+
+Opening this url in a browser will result in a redirect, which lands you on the test plugin ui:
+
+![Browser showing communication test plugin ui](/img/setup_complete.png)
+
+Meanwhile, the communication test plugin will have logged the decoded authentication result, which will look like
+```
+Decoded: AuthResult { status: Succes, attributes: Some({"test_attribute": "test_attribute_value"}), session_url: None }
+```
+
+Congratulations! Seeing the above indicates that your setup now works correctly. You can experiment further by looking at the configuration options described in (TODO) and (TODO).
```

Comment:
Do we need to do the todo's? Or do we do the todo's later

Reply:
woeps, forgot bout those. Will fix